src/app/design/frontend/Scandiweb/pwa/src/app/route/HomePage/HomePage.component.js

```diff
@@ -1,27 +1,102 @@
 import { PureComponent } from 'react';
-import CmsPage from 'Route/CmsPage';
+import PropTypes from 'prop-types';
 import GenderSlider from 'Component/GenderSlider';
+import SliderVerticalWidget from 'Component/SliderVerticalWidget';
+import { backIcon } from 'Component/Header/Header.config';
+import { WOMEN, MEN } from 'Component/GenderSlider/GenderSlider.component';
 
 import './HomePage.style.scss';
 
+const WOMEN_SLIDER_ID = 5;
+const MEN_SLIDER_ID = 6;
+
 export default class HomePage extends PureComponent {
     static propTypes = {
+        genderSwitchIndex: PropTypes.number.isRequired
     };
 
     static defaultProps = {
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            activeSlide: {
+                [WOMEN]: 0,
+                [MEN]: 0
+            },
+            slideCount: 0
+        };
+
+        this.handleArrowClick = this.handleArrowClick.bind(this);
+        this.getSlideCount = this.getSlideCount.bind(this);
+    }
+
+    getSlideCount(slideCount) {
+        this.setState({ slideCount });
+    }
+
+    handleArrowClick() {
+        const { genderSwitchIndex } = this.props;
+        const { activeSlide, activeSlide: { [genderSwitchIndex]: activeSlideIndex }, slideCount } = this.state;
```

Comment:
prefer each on new line, e.g.

```suggestion
        const {
                activeSlide, 
                activeSlide: { [genderSwitchIndex]: activeSlideIndex },
                slideCount
        } = this.state;
```